### PR TITLE
streamingest: allow replicating system tenants to other tenants

### PR DESCRIPTION
**Pull request by dt**
Release note: none.
Epic: https://cockroachlabs.atlassian.net/browse/CRDB-33245

### CODE REVIEW

**Comment by dt** [comment]
Now with a couple tests and RFAL.

File: pkg/ccl/streamingccl/replicationtestutils/testutils.go [hunk 1]
```diff
@@ -142,11 +142,13 @@ func (c *TenantStreamingClusters) setupSrcTenant() {
...
-	require.NoError(c.T, c.SrcCluster.Server(0).TenantController().WaitForTenantCapabilities(ctx, c.Args.SrcTenantID, map[tenantcapabilities.ID]string{
-		tenantcapabilities.CanUseNodelocalStorage: "true",
-	}, ""))
+	if !c.Args.SrcTenantID.IsSystem() {
+		c.SrcSysSQL.Exec(c.T, `ALTER TENANT $1 SET CLUSTER SETTING sql.virtual_cluster.feature_access.multiregion.enabled=true`, c.Args.SrcTenantName)
+		c.SrcSysSQL.Exec(c.T, `ALTER TENANT $1 GRANT CAPABILITY can_use_nodelocal_storage`, c.Args.SrcTenantName)
+		require.NoError(c.T, c.SrcCluster.Server(0).TenantController().WaitForTenantCapabilities(ctx, c.Args.SrcTenantID, map[tenantcapabilities.ID]string{
+			tenantcapabilities.CanUseNodelocalStorage: "true",
+		}, ""))
+	}
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/replicationtestutils/testutils.go:151]: Unrelated follow-up: I wonder how much of this we need given that this uses shared service tenants and these should be the defaults now.

**Review by stevendanna — APPROVED** [review]
LGTM but perhaps we should let @msbutler have a look as well since I already watched this code get written and might be a bit biased.

File: pkg/ccl/streamingccl/streamproducer/producer_job.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ import (
 )
 
 func makeTenantSpan(tenantID uint64) roachpb.Span {
+	// TODO(dt): remove conditional if we make MakeTenantSpan do this.
```
Comment (msbutler) [inline pkg/ccl/streamingccl/streamproducer/producer_job.go:37]: could you add a docstring which explains why this is necessary? 

Also, for a regular main tenant, do they have any non table data in their key space? I.e. if we replicate the key span `keys.MakeTenantSpan(3)`, will that only ever contain table data?

**Review by msbutler — APPROVED** [review]
Nice! I'm glad all the span config streaming stuff works out of the box! Were you able to verify that the a user connected to the system tenant cannot directly write to the zone config table?

Also, it may be helpful to write out in the commit message some of the subtle ways that streaming from the system tenant is different than streaming from a main tenant (e.g. we'll replicate a zone configs table that will never get used after cutover).

File: pkg/ccl/streamingccl/streamproducer/producer_job.go (hunk 2 above)
Comment (dt) [reply pkg/ccl/streamingccl/streamproducer/producer_job.go:37]: > or a regular main tenant, do they have any non table data in their key space

No.

> replicate the key span keys.MakeTenantSpan(3), will that only ever contain table data

Correct.

**Comment by dt** [comment]
>   the subtle ways that streaming from the system tenant is different... we'll replicate a zone configs table that will never get used after cutover

I think we overstated those differences in our chat; e.g. the zones table _does_ get used, that is read by the reconciler and fed to span configs. After #122005 there really aren't many differences. It really is just that some jobs are only supposed to run in the system tenant so their owners may need to be aware of the fact they could end up in a non-system tenant and need to terminate themselves, a la the first commit here where we do that for our pcr job.

**Pull request merged**